.github/workflows/unit-tests.yml

```diff
@@ -0,0 +1,105 @@
+name: Unit Tests
```

Comment:
Better name here... "Build and Test", "CI Tests"??

Reply:
i decided not to name it "build and test", since that's the name of one of the jobs (build_and_test and check_code). ill switch to CI Tests if that works with you